datacube/metadata/_eo3converter.py

```diff
@@ -161,14 +161,52 @@ def _compute_uuid(
 
 
 def _to_grid(gbox: GeoBox) -> dict[str, Any]:
-    return {"shape": gbox.shape.yx, "transform": gbox.transform[:6]}  # type: ignore[index]
+    return {"shape": list(gbox.shape.yx), "transform": list(gbox.transform[:])}  # type: ignore[index]
```

Comment:
On the micro-level: when is `gbox.transform[:]` different from `gbox.transform`?

On a higher level: if the previous slice isn't required any longer, can't the list conversion here be removed and "transform" remain an `Affine`?

Reply:
There's no meaningful difference; I kept the list slicing because `list(gbox.transform)` gave a mypy error (No overload variant of "list" matches argument type "Affine") - then again, mypy errors on this too.
I wanted to convert it to a list to be consistent with how the value appears in metadata yamls. Beyond that, I'm not sure if there are any potential impacts to having the transform value be an Affine object rather than a list.

Reply:
I had a look and Affine is a class inheriting from `namedtuple` of 9 components. Are you trying to fix some bug and deliberately change what `transform` is in this PR (so it includes all 9 components instead of just 6 as earlier), or is this change an outcome of fighting the type checker?

I don't know what things should be, I'm just reacting to things in the code that surprised me.